src/core/client_channel/retry_filter_legacy_call_data.cc

```diff
@@ -1348,7 +1348,8 @@ void RetryFilter::LegacyCallData::CallAttempt::BatchData::
       calld->send_messages_[call_attempt_->started_send_message_count_];
   ++call_attempt_->started_send_message_count_;
   batch_.send_message = true;
-  batch_.payload->send_message.send_message = cache.slices;
+  batch_.payload->send_message.send_message =
+      calld->arena_->New<SliceBuffer>(cache.slices->Copy());
```

Comment:
Allocating this via `Arena::New()` means that the dtor will never be called.  @ctiller can correct me if there's some reason to think that's okay, but I would assume it's not.

I think the right way to solve this is to add a `SliceBuffer` data member to `CallAttempt`.  There can be only one send_message op in flight on the call attempt at any given time, so the code here can simply copy the slices from the cache to the slice buffer data member, and then pass a pointer to that data member down in the payload.  We probably also need to clear the slice buffer [when the send_message op completes](https://github.com/grpc/grpc/blob/2eaaa9cbf7226054df83b19ddd6d530a2e8edb32/src/core/client_channel/retry_filter_legacy_call_data.cc#L1270).

Reply:
done. thanks! ASAN also caught the memory leak since the destructor of SliceBuffer was not called.